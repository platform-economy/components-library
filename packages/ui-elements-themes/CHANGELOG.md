# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
