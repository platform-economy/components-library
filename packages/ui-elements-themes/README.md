# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
