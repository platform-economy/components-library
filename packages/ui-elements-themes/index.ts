export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
