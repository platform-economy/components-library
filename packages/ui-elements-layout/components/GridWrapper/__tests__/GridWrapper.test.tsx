import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';
import { withTestTheme } from '@relayr/ui-elements-themes';

// Component
import { GridWrapper, GridGrow, columnPercentage, growCounter } from '../index';

describe('ui-elements-layout', () => {
  describe('Components', () => {
    describe('GridWrapper', () => {

      it('should render with Grid Grow element and correct props', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <GridWrapper columns={2}>
              <GridGrow columns={2} colspan={1}>
                <div />
              </GridGrow>
            </GridWrapper>))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render with Grid Grow element and incorrect props', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <GridWrapper columns={2}>
              <GridGrow columns={2} colspan={0}>
                <div />
              </GridGrow>
            </GridWrapper>))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('counts proper column width and colspan', () => {
        const props = { columns: 3, colspan:2 };

        expect(columnPercentage(props.columns)).toBe(32);
        expect(growCounter(props.columns, props.colspan)).toBe(66);
      });

      it('counts proper column width with one column', () => {
        const props = { columns: 1 };

        expect(columnPercentage(props.columns)).toBe(100);
      });

      it('counts proper columns width when colspan is incorrect', () => {
        const props = { columns: 3, colspan:0 };

        expect(growCounter(props.columns, props.colspan)).toBe(32);
      });

      it('counts proper column width with incorrect column props', () => {
        const props = { columns: 7 };

        expect(columnPercentage(props.columns)).toBe(100);
      });
    });
  });
});
