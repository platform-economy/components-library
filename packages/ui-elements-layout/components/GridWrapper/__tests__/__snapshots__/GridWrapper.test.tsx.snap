// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-layout Components GridWrapper should render with Grid Grow element and correct props 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > div {
  box-sizing: border-box;
  -webkit-flex-basis: 49%;
  -ms-flex-preferred-size: 49%;
  flex-basis: 49%;
  margin-bottom: 14px;
}

.c0 > :first-child {
  margin-left: 0;
}

.c0 > :last-child {
  margin-right: auto;
  margin-left: 2%;
}

.c1.c1 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-basis: 49%;
  -ms-flex-preferred-size: 49%;
  flex-basis: 49%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div />
  </div>
</div>
`;

exports[`ui-elements-layout Components GridWrapper should render with Grid Grow element and incorrect props 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > div {
  box-sizing: border-box;
  -webkit-flex-basis: 49%;
  -ms-flex-preferred-size: 49%;
  flex-basis: 49%;
  margin-bottom: 14px;
}

.c0 > :first-child {
  margin-left: 0;
}

.c0 > :last-child {
  margin-right: auto;
  margin-left: 2%;
}

.c1.c1 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-basis: 49%;
  -ms-flex-preferred-size: 49%;
  flex-basis: 49%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div />
  </div>
</div>
`;
